### DISCUSSION TITLE
请问下，apisix init 的时候报错，用的 centos7.9

### DISCUSSION THREAD

**Discussion by woyouyitouchenfeijv** [apache/apisix#6342] [Q&A] [answered]
在apisix包外执行apisix init，报错是这样
Linux version 3.10.0-1160.el7.x86_64 (mockbuild@kbuilder.bsys.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) ) #1 SMP Mon Oct 19 16:18:59 UT
C 2020
apisix init err:

/usr/local/openresty/luajit/bin/luajit: /usr/local/apisix/apisix/cli/ops.lua:435: bad argument #1 to 'ipairs' (table expected, got nil)
stack traceback:
        [C]: in function 'ipairs'
        /usr/local/apisix/apisix/cli/ops.lua:435: in function </usr/local/apisix/apisix/cli/ops.lua:149>
        /usr/local/apisix/apisix/cli/ops.lua:864: in function 'execute'
        /usr/local/apisix/apisix/cli/apisix.lua:34: in main chunk
        [C]: at 0x00405260

在apisix包里执行报错
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init
/usr/local/openresty/luajit/bin/luajit: error loading module 'socket.core' from file '/opt/csswebapi/apisix/deps/lib/lua/5.1/socket/core.so':
        /opt/csswebapi/apisix/deps/lib/lua/5.1/socket/core.so: cannot open shared object file: No such file or directory
stack traceback:
        [C]: at 0x00454380
        [C]: in function 'require'
        /opt/csswebapi/apisix/deps/share/lua/5.1/socket.lua:12: in main chunk
        [C]: in function 'require'
        /opt/csswebapi/apisix/deps/share/lua/5.1/socket/http.lua:10: in main chunk
        [C]: in function 'require'
        ./apisix/cli/etcd.lua:22: in main chunk
        [C]: in function 'require'
        ./apisix/cli/ops.lua:18: in main chunk
        [C]: in function 'require'
        ./apisix/cli/apisix.lua:32: in main chunk
        [C]: at 0x00405260

提示说是找不到core.so，但是这个文件是存在的。

**Comment by tokers** [comment, marked as answer]
How did you install APISIX?

What's the version of APISIX?

**Reply by woyouyitouchenfeijv** [reply]
谢谢了。🙏 
我用的是2.12.1。

最后找到问题了，是因为openresty 和 openssl 用的是arm的安装，而机器是x86的。

**Comment by leslie-tsang** [comment]
Hello there, could you please explain how you installed APISIX?
> 提示说是找不到core.so，但是这个文件是存在的。

Please check the permission.

**Reply by woyouyitouchenfeijv** [reply]
嗯嗯。谢谢了
是我按错了openresty 和 openssl 的版本，应该用和服务器相同架构的，我选错了，x86的选了arm的，安装openresty 和 openssl没出问题，但是apisix在安装的时候就出错了。

最后重新安了openresty 和 openssl 就OK了
